### PR TITLE
forceESM for vitest

### PR DESCRIPTION
**Pull request by Uzlopak**
Alternative for #273 

@climba03003 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — APPROVED** [review]
This one certainly better.

**Comment by Uzlopak** [comment]
Tbh i personally dont understand why forcing esm avoids typescript loading?! :D

File: vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    include: ['test/vitest/**/*.test.ts']
+    // ...
```
Comment (Uzlopak) [inline vitest.config.ts:6]:
```suggestion
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**